### PR TITLE
goal: Rename source map output file

### PR DESCRIPTION
**Pull request by algochoi**
## Summary

Follow up to #3726 - renamed source map file to match the output file rather than the input file to follow [existing conventions ](https://sourcemaps.info/spec.html) more closely. Thanks to @jasonpaulos for the catch!

## Test Plan

Changed an existing `e2e` test to match the new file path.

### CODE REVIEW

File: cmd/goal/clerk.go [hunk 1]
```diff
@@ -1070,7 +1070,7 @@ var compileCmd = &cobra.Command{
 				}
 			}
 			if writeSourceMap {
-				mapname := fname + ".map"
+				mapname := outname + ".map"
```
Comment (jasonpaulos) [inline cmd/goal/clerk.go:1073]: What if `outname` is `stdoutFilenameValue`, which is "-"? I think `fmt.Sprintf("%s.tok.map", fname)` would be a better choice in that case

**Review by jasonpaulos — COMMENTED** [review]
Just have one edgecase comment, otherwise looks good!

File: cmd/goal/clerk.go (hunk 1 above)
Comment (algochoi) [reply cmd/goal/clerk.go:1073]: Good catch, I'll change this - what is the purpose of having a `stdoutFilenameValue`?

Comment (jasonpaulos) [reply cmd/goal/clerk.go:1076]: Oh wait my suggestion is broken for this reason: if `outname` is `stdoutFilenameValue`, then `fname` _must_ be `stdinFileNameValue`, which is also "-".

Comment (jasonpaulos) [reply cmd/goal/clerk.go:1076]: I guess the only thing you can do is error?

The situation would only happen if someone is piping into goal and out of it while also specifying maps should be generated, e.g. `cat ... > goal clerk compile -m > some_other_command`. It appears to me there's no intuitive place for the maps to go.

Comment (algochoi) [reply cmd/goal/clerk.go:1076]: Ah okay, I see why it's `-` now (literally for stdin/stdout). I'll error out in this case for now (but I wouldn't be opposed to printing it to stdout either).

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**